Sources/PubNub/EventEngine/Core/EventEngine.swift

```diff
@@ -66,7 +66,13 @@ class EventEngine<State, Event, Invocation: AnyEffectInvocation, Input> {
     }
 
     let transitionResult = transition.transition(from: currentState, event: event)
+    let newState = transitionResult.state
     let invocations = transitionResult.invocations
+    let currentStateDescr = String(describing: currentState)
+    let newStateDescr = String(describing: newState)
+
+    PubNub.log.debug("Exiting \(currentStateDescr)", category: LogCategory.eventEngine.rawValue)
```

Comment:
I was curious about performance hit because of formatting which won't be used if logger is off. Should we have some kind of closure in this function which conditionally will be called?

Reply:
This is how exactly it should work under the hood. The `message` parameter for the `PubNub.log.debug` and simillar methods as well uses autoclosure, which means that a value evaluation will be deferred if needed. Anyway, I will double check it if it works as expected and let you know